Beam.Server.Application/Services/Interfaces/IUserService.cs

```diff
@@ -0,0 +1,17 @@
+﻿using Beam.Infrastructure.Entities;
+
+
+namespace Beam.Application.Services.Interfaces;
+
+public interface IUserService
+{
+    Task<List<User>> GetAllAsync();
+    
+    Task<User> GetByIdAsync(Guid id);
+    
+    Task<Guid> CreateAsync(User input);
```

Comment:
Entities should not appear outside of Application layer. Create folder `Dto` and put there two classes:
UserDto, that stores User's data as method Response:
```
public class UserDto
{
    public Guid Id { get; set; }
    public string Tag { get; set; }
    public string Name { get; set; }
    public DateTimeOffset CreationDate { get; set; }
}
```
UserInputDto, that stores User's data as method Request:
```
public class UserDto
{
    public string Tag { get; set; }
    public string Name { get; set; }
    public string Password { get; set; }
}
```

For example, while creating User we send UserInputDto, but get UserDto. So the method should look like:
```
Task<UserDto> CreateAsync(UserInputDto input);
```